INSTALL

```diff
@@ -214,4 +228,82 @@ all images on a storage device such as eMMC or SD card.
+the latter will do all required actions automatically. All what we need to care about is to write
+proper environment variables from U-Boot command line (boot_dev/bootcmd) according to the chosen SDx.
+See workspace_base/meta-xt-prod-gen3-test/doc/u-boot-env-salvator-x.txt for details.
+
+1.2 In case of eMMC booting, we have to have an access to it in order to flash images.
+It is going to be not quite easy as for removable SD card, but the one of the possible ways is
+to prepare the image blob using the same script, copy resulting blob to NFS root directory,
+set system to boot via NFS, go to a DomD on target (where eMMC device is available)
+and using "dd" command just copy blob to eMMC.
+
+For example,
+
+prepare an image blob:
+sudo ./mk_sdcard_image.sh -p /IMAGE_FOLDER -d /home/emmc.img
+
+and then run on target:
+dd if=/home/emmc.img of=/dev/mmcblk0
+
+After getting eMMC flashed we have to choose it to be an boot device in a similar way as it is done
+for SD card.
+See u-boot-env-salvator-x.txt for details.
+
+2. Using TFTP boot with NFS root.
+In case of TFTP booting we have to copy the following boot images into target-accessible TFTP boot
+directory and extract guest domain rootfs images for being an NFS root directories for guest domains.
+See https://github.com/xen-troops/manifests/wiki on how to setup TFTP, NFS, etc.
+
+For example,
+
+copy boot images to TFTP boot directory:
+sudo mkdir /srv
+cd workspace_base/build/tmp/deploy/dom0-image-thin-initramfs/images/generic-armv8-xt/
+sudo cp uInitramfs /srv/
+sudo cp Image /srv/
+cd workspace_base/build/tmp/deploy/domd-image-weston/images/MACHINE_NAME-xt/
+sudo cp dom0.dtb /srv/
+sudo cp xenpolicy /srv/
+sudo cp xen-uImage /srv/
+
+extract DomD rootfs images:
+sudo mkdir /srv/domd
+cd workspace_base/build/tmp/deploy/domd-image-weston/images/MACHINE_NAME-xt/
+sudo tar -xf core-image-weston-MACHINE_NAME-xt.tar.bz2 -C /srv/domd/
+
+extract DomU rootfs images:
+sudo mkdir /srv/domu
+cd workspace_base/build/tmp/deploy/domu-image-weston/images/MACHINE_NAME-xt/
+sudo tar -xf core-image-weston-MACHINE_NAME-xt.tar.bz2 -C /srv/domu/
+
+It worth mentioning that some changes to domain config files are necessary to switch between NFS
+and any of storage devices.
+
+For doing that:
+cd workspace_base/meta-xt-prod-gen3-test/recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/
+
+- Edit domd-MACHINE_NAME.cfg to uncomment "extra" option which corresponds to NFS.
+- Edit domu-MACHINE_NAME.cfg to uncomment "extra" option which corresponds to NFS and comment "disk" option.
+
+So, as you can see, varying U-Boot's "boot_dev" and "bootcmd" environment variables
+and domain config's "extra" and "disk" options it is possible to choose  different boot device
+for each system component. For example, boot system using TFTP and then use SD card for guest domains.
+
+Remark regarding uirfs.sh script.
```

Comment:
I guess remarking about something requires this to be mentioned before.

Reply:
ok. I will explain